File: iot/mqttsn/sn_messages/SNPubrec.py
import iot.mqttsn.sn_classes.SNmessageType as SNmessageType

class snPubrec():
    def __init__(self, packetID):
        self.packetID = packetID
        self.snMessageType = SNmessageType.snMessageType()

    def getLength(self):
        return 4

    def getType(self):
        return self.snMessageType.getValueByKey('SN_PUBREC')

    def getPacketID(self):
        return self.packetID

    def setPacketID(self, packetID):
        self.packetID = packetID

File: IoTbrokerClient/src/iot-protocols/mqtt-sn/messages/snpubrec.cpp
/**
 * Mobius Software LTD
 * Copyright 2015-2018, Mobius Software LTD
 *
 * This is free software; you can redistribute it and/or modify it
 * under the terms of the GNU Lesser General Public License as
 * published by the Free Software Foundation; either version 2.1 of
 * the License, or (at your option) any later version.
 *
 * This software is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public
 * License along with this software; if not, write to the Free
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
 */

#include "snpubrec.h"

SNPubrec::SNPubrec()
{

}

SNPubrec::SNPubrec(int packetID)
{
    this->packetID = packetID;
}

int SNPubrec::getLength()
{
    return 4;
}

int SNPubrec::getType()
{
    return SN_PUBREC;
}

IotEnumProtocol *SNPubrec::getProtocol()
{
    return new IotEnumProtocol(MQTT_SN_PROTOCOL);
}

int SNPubrec::getPacketID()
{
    return this->packetID;
}

void SNPubrec::setPacketID(int value)
{
    this->packetID = value;
}
